app/controllers/ops_controller/settings/upload.rb

```diff
@@ -90,4 +88,12 @@ def upload_csv
     session[:flash_msgs] = @flash_array.dup if @flash_array
     redirect_to(:action => 'explorer', :no_refresh => true)
   end
+
+  private
+
+  def logo_dir
+    dir = File.expand_path(Rails.root.join('public', 'upload'))
```

Comment:
You can use the Pathname object directly instead of relying on the File class methods

```ruby
dir = Rails.root.join('public', 'upload').expand_path
Dir.mkdir(dir) unless dir.exist?
dir.to_s
```